src/pages/index.js

```diff
@@ -1,90 +1,266 @@
+    userInfo.setUserInfo({
+      name: profileData.name,
+      about: profileData.about,
+    });
+    userInfo.setUserAvatar({ avatar: profileData.avatar });
+  })
+  .catch((err) => console.error(err));
+
+/* PATCH Profile Edit Function */
+function handleProfileEditSubmit(inputValues) {
+  editProfileModal.setSubmitButtonText("Saving...");
+
+  if (!inputValues.name || !inputValues.about) {
+    console.error("Please fill in all the fields");
+    return;
+  }
+
+  api
+    .editProfile({
+      name: inputValues.name,
+      about: inputValues.about,
+    })
+    .then(({ name, about }) => {
+      userInfo.setUserInfo({
+        name: name,
+        about: about,
+      });
+      editProfileModal.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info:", err);
+    })
+    .finally(() => {
+      editProfileModal.setSubmitButtonText("Save");
+    });
+}
+
+/* PATCH Profile Avatar Function */
+function handleProfileImageSubmit(inputValues) {
+  updateProfileImage.setSubmitButtonText("Saving...");
+
+  api
+    .editProfileImage(inputValues)
+    .then((res) => {
+      userInfo.setUserAvatar(res);
+      updateProfileImage.reset();
+      updateProfileImage.close();
+      console.log("Success:", res);
+    })
+    .catch((err) => {
+      console.error("Error updating profile avatar:", err);
+    })
+    .finally(() => {
+      updateProfileImage.setSubmitButtonText("Save");
+    });
+}
+
+/* POST Add Place Function */
+function handleNewPlaceSubmit(inputValues) {
+  addPlaceModal.setSubmitButtonText("Saving...");
+
+  api
+    .addCard({ name: inputValues.name, link: inputValues.link })
+    .then(({ name, link, _id }) => {
+      section.addItem(getCardElement({ name, link, _id }));
+      addPlaceModal.close();
+
+      addPlaceModal.reset();
+    })
+    .catch((err) => {
+      console.error("Error adding new place:", err);
+    })
+    .finally(() => {
+      addPlaceModal.setSubmitButtonText("Save");
+    });
+}
+
+/* PUT Add Like React */
+function handleLikeReact(card) {
+  if (card.isLiked) {
+    api
+      .removeLikeReact(card._id)
+      .then(() => {
+        card.toggleLike();
+        card.isLiked = false;
+      })
+      .catch((error) => console.error("Error removing like reaction:", error));
+  }
+  if (!card.isLiked) {
+    api
+      .addLikeReact(card._id)
+      .then(() => {
+        card.toggleLike();
+        card.isLiked = true;
+      })
+      .catch((error) => console.error("Error adding like reaction:", error));
+  }
+}
+
+/* Edit Profile Button Listener */
+profileEditBtn.addEventListener("click", () => {
+  const currentUser = userInfo.getUserInfo();
+  profileTitleInput.value = currentUser.name || "";
+  profileDescriptionInput.value = currentUser.about || "";
+  profileEditValidation.resetValidation();
+  editProfileModal.open();
+});
+
+/* Add Place Button Listener */
+placesAddBtn.addEventListener("click", () => {
+  addPlaceModal.open();
+});
+
```

Comment:
Needs correction: the delete confirmation popup disappears right after opening it, but you should confirm the card removing manually